proposals/params-span.md

```diff
@@ -0,0 +1,146 @@
+# `params Span<T>` and implicit allocation of arrays
+
+## Summary
+Avoid heap allocation for implicit allocation of arrays in specific scenarios such as `params` arguments.
```

Comment:
```suggestion
Avoid heap allocation for implicit allocation of arrays in specific scenarios with `params` arguments.
```

Reply:
The motivation for suggestion is that some `params Span<T>` scenarios are likely to still allocate.